eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Utilities/ContextualParameterBuilder.cs

```diff
@@ -57,6 +61,12 @@ private static void BuildContextualParameterHierarchy(RequestPathPattern current
                 // using the reference name of the last segment as the parameter name, aka resourceGroupName
                 parameterStack.Push(new ContextualParameter(current[^2].Value, current[^1].VariableName, id => id.ResourceGroupName()));
             }
+            else if (current == RequestPathPattern.Extension)
+            {
+                Func<ScopedApi<ResourceIdentifier>, ScopedApi<ResourceIdentifier>> idExpression =
+                    buildingScope == ContextParameterBuildingScope.ResourceCollection ? id => id : id => id.Parent();
+                parameterStack.Push(new ContextualParameter("resourceUri", "resourceUri", idExpression));
```

Comment:
I think this is incorrect.
When the input is `/{resourceUri}`, the variable expression should always just be `Id`.
We should never need this `ContextParameterBuildingScope`.

Can you add some more unit test to this class to verify this?

Reply:
Indeed, the "buildingScope" thing is unnecessary.